src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.security.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+import org.apache.commons.validator.routines.EmailValidator;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        String emailNoWhitespace = email.strip();
+        String passwordNoWhitespace = password.strip();
+        Optional<User> userOptional = userService.findByEmail(emailNoWhitespace);
+        User user = userOptional.orElseThrow(
+                () -> new AuthenticationException("User or password is incorrect"));
+        String hashedPassword = HashUtil.hashPassword(passwordNoWhitespace, user.getSalt());
+        if (user.getPassword().equals(hashedPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("User or password is incorrect");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        String emailNoWhitespace = email.strip();
+        String passwordNoWhitespace = password.strip();
+        if (!EmailValidator.getInstance().isValid(emailNoWhitespace)) {
+            throw new RegistrationException("Email '" + emailNoWhitespace + '\'' + " is not valid");
+        }
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException(
+                    "User with email '" + emailNoWhitespace + '\'' + " is already exist");
+        }
+        if (passwordNoWhitespace.isEmpty()) {
+            throw new RegistrationException("Password could not be empty");
+        }
```

Comment:
Password validation is out of scope. Keep in mind KISS principle.

Reply:
Are you proposing to handle blanks or whitespace characters for input password? Or maybe the password check would be in the app somewhere up front? Explain, please. I would like to understand this issue. As I understand the principle of KISS is about simplicity, but not at the cost of omissions and flaws.
The Common mistakes page (https://mate-academy.github.io/jv-program-common-mistakes/hibernate/add-user-service/add-user-service-checklist) recommends using password.isEmpty(). For what then?